src/main/scala/medeia/decoder/BsonDecoder.scala

```diff
@@ -102,6 +105,19 @@ trait DefaultBsonDecoderInstances extends BsonIterableDecoder {
 
   implicit def chainDecoder[A: BsonDecoder]: BsonDecoder[Chain[A]] = listDecoder[A].map(Chain.fromSeq)
 
+  implicit def mapDecoder[K: BsonKeyDecoder, A: BsonDecoder]: BsonDecoder[Map[K, A]] = bson => {
+    val document = bsonDocumentDecoder.decode(bson)
+    document.flatMap { doc: BsonDocument =>
+      val x: EitherNec[BsonDecoderError, List[(K, A)]] = doc.asScala.toList.parTraverse {
```

Comment:
is this `x` variable a leftover?

Reply:
> However in our case it's not _really_ commutative anyway right? The result is different in the sense that the order of errors in the `Chain` does change and so the laws would not hold. At least in my head I think it would be possible to use it if we switch out the `Chain` for e.g. a `Set` but IMHO it's not worth the pain currently ;)

Yes, I think that is  right. Not worth the pain :) Also because I believe that the order inside the bson documents are preserved and therefore the errors kind of have an order, even if the interface of Map does not guarantee it. 

I was wondering why unorderedParTraverse was missing in cats. To me it seems "incomplete" maybe I will ask at their gitter.

Reply:
Is a leftover, will remove it.